webapp/src/client/client.ts

```diff
@@ -15,6 +17,11 @@ class SurveyClient extends HttpClient {
     doConnected = async () => {
         return this.doPost(`${this.url}/connected`);
     };
+
+    submitSurveyResponse = async (surveyID: string, response: SurveyResponse) => {
+        const url = `${this.url}/survey/${surveyID}/response`;
```

Comment:
I saw the change on the server side to verify the survey id, but this could still potentially do something like passing a server id like `../../../some/dangerous/route?foo=` which will end in `myinstance.com/plugins/pluginId/survey/../../../some/dangerous/route?foo=response`.

We should do some validation of the survey ids before calling this function, or be more careful when building this path.

Reply:
No, doing so results in 404.

Reply:
@hmhealey If I am not mistaken, I can use the API to create a new post with props equal to the plugin's and set the Survey ID in any way I want. Since the rendering is also not limited to the bot chat, this may end up in using the constructed ID.

Reply:
@larkox if such a duplicate post contains a non-existing ID, nothing would happen on the server side other than a few DB calls. I don't see any concern there.

Reply:
Whats the solution that you propose. And I'm considering this as a non-blocking concern for this PR.

Reply:
Discussed offline. To be handled in a different PR.

Probably the best approach is to verify the id match a regex similar to the ones we use in webapp here: `webapp/channels/src/utils/path.ts`

```
export const ID_PATH_PATTERN = '[a-z0-9]{26}';
```

Reply:
Done.